swan-lake/learn-the-platform/test-document-the-code/debug-ballerina-programs.md

```diff
@@ -168,6 +168,22 @@ Follow the steps below to start a remote debug session.
 
 <br/>
 
+### Run the Ballerina program in a separate terminal while debugging
+
+Follow the steps below to start a program debug session that launches the Ballerina program in a separate integrated terminal. This would make it possible to debug programs that take user input in the **Ballerina Debug** mode.
+
+1. Create the `launch.json` configuration file if it is not created already. For instructions on creating the `launch.json` file, see [start a debug session with configurations](#start-a-debug-session-with-configurations).
+
+2. Open the `launch.json` file, and set the `terminal` configuration under the `Ballerina Debug` section to `integrated`. 
+
+3. After setting the configuration, click the **Start Debugging** icon on the upper left corner to start debugging.
+
+    You will see the program executing on a separate integrated terminal titled **BALLERINA DEBUG TERMINAL**. Simultaneously, you can use the **DEBUG CONSOLE** for evaluations.
+
+    ![Debug Run Program in Terminal](/learn/images/debugger-run-in-terminal.gif)
```

Comment:
IMO we can have this content under the `Ballerina `launch.json` attributes` section, as from the user's POV this an additional config introduced for the debugger. WDYT?

Reply:
+1. Will update accordingly.